src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -511,13 +512,25 @@ export class VisualizeEmbeddable
         layers: [] as VisLayers,
       };
       // We cannot use this.handler in this case, since it does not support the run() cmd
-      // we need here. So, we consume the expressions service to run this instead.
+      // we need here. So, we consume the expressions service to run this directly instead.
       const exprVisLayers = (await getExpressions().run(
         visLayersPipeline,
         visLayersPipelineInput,
         expressionParams as Record<string, unknown>
       )) as ExprVisLayers;
-      return exprVisLayers.layers;
+      const visLayers = exprVisLayers.layers;
+      const err = getAnyErrors(visLayers, this.vis.title);
+      if (err !== undefined) {
+        const { toasts } = getNotifications();
+        toasts.addError(err, {
+          title: i18n.translate('visualizations.renderVisTitle', {
+            defaultMessage: `Error loading data on the ${this.vis.title} chart`,
+          }),
+          toastMessage: ' ',
```

Comment:
why is the toast empty? Can you add a screenshot of the toast to the PR? it will be helpful to see what this looks like. Also you might want to add an id to the toast if you want to prevent he toast from being displayed again should the same expression run again before the toast is dismissed (But be careful to keep the ID specific to he vis since the id is global to the dashboard)

Reply:
The reason for the toast message being empty here is because it is reserved for showing up in the modal details (the danger callout text). Added a screenshot of the toast to description.

I'll add an ID to the toast after rebasing and pulling in the upstream changes that include the ID as an option.

Note this is all happening inside an `if (err !== undefined)` block. This is only true when one or more VisLayers had an error. I'll add a comment in there too.

Reply:
Update: after checking #3752 it doesn't include updating `ErrorToastOptions` which is needed for adding an ID to `addError` fn used for displaying error toasts. I will leave this empty for now and we can add later if that is decided to be added.

Reply:
I've added the ID so it can be handled for `addError` scenarios. See commit https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3649/commits/739ddaa7d2bb24557b22e4200649eaaebefc62ff

Reply:
Is the ID here coming from the plugin resource ID?

Reply:
Using the visualize_embeddable ID such that it's unique within the dashboard. This enforces that there will only be one toast per vis on a dashboard.

Reply:
Did you do any testing what the toast would look like or have seen before if we have dozens of detector per the viz and they all have some error, will the toast be scrollable at some point?

Reply:
I haven't individually tested that but yes, it will be scrollable. It is designed to hold very long text (error stack traces)

Reply:
Sorry to reopen the thread, I'm just trying to understand whether our toast interface needs to be improved. The mocks make sense to me, where we have only a title in the toast. But why do we need to pass a whitespace character as `toastMessage`? It seems like we should be able to omit that property altogether, or else pass undefined, null, or at least an empty string. Do we need to open an issue to improve the `addError` method?